token/program-2022-test/tests/confidential_transfer.rs

```diff
@@ -199,39 +273,349 @@ async fn ct_enable_disable_balance_credits() {
+        )
+        .await
+        .unwrap();
+
+    let state = token
+        .get_account_info(&alice_meta.token_account)
+        .await
+        .unwrap();
+    let extension = state
+        .get_extension::<ConfidentialTransferAccount>()
+        .unwrap();
+    assert_eq!(
+        extension.decryptable_available_balance,
+        new_decryptable_available_balance.into(),
+    );
+    assert_eq!(extension.pending_balance_credit_counter, 1.into());
+    assert_eq!(extension.expected_pending_balance_credit_counter, 1.into());
+    assert_eq!(extension.actual_pending_balance_credit_counter, 1.into());
+    assert_eq!(
+        extension.pending_balance,
+        zk_token_elgamal::pod::ElGamalCiphertext::zeroed()
+    );
+    assert_ne!(
+        extension.available_balance,
+        zk_token_elgamal::pod::ElGamalCiphertext::zeroed()
+    );
+}
+
+#[tokio::test]
+async fn ct_withdraw() {
+    let ConfidentialTransferMintWithKeypairs { ct_mint, .. } =
+        ConfidentialTransferMintWithKeypairs::new();
+    let mut context = TestContext::new().await;
+    context
+        .init_token_with_mint(vec![
+            ExtensionInitializationParams::ConfidentialTransferMint { ct_mint },
+        ])
+        .await
+        .unwrap();
+
+    let TokenContext {
+        token,
+        alice,
+        mint_authority,
+        decimals,
+        ..
+    } = context.token_context.unwrap();
+    let alice_meta =
+        ConfidentialTokenAccountMeta::with_tokens(&token, &alice, &mint_authority, 42, decimals)
+            .await;
+
+    let state = token
+        .get_account_info(&alice_meta.token_account)
+        .await
+        .unwrap();
+    assert_eq!(state.base.amount, 0);
+
+    token
+        .confidential_transfer_withdraw(
+            &alice_meta.token_account,
+            &alice_meta.token_account,
+            &alice,
+            21,
+            decimals,
+            42,
+            &alice_meta.elgamal_keypair,
+            alice_meta.ae_key.encrypt(42_u64 - 21),
+        )
+        .await
+        .unwrap();
+
+    let state = token
+        .get_account_info(&alice_meta.token_account)
+        .await
+        .unwrap();
+    assert_eq!(state.base.amount, 21);
+
+    token
+        .confidential_transfer_withdraw(
+            &alice_meta.token_account,
+            &alice_meta.token_account,
+            &alice,
+            21,
+            decimals,
+            21,
+            &alice_meta.elgamal_keypair,
+            alice_meta.ae_key.encrypt(0u64),
+        )
+        .await
+        .unwrap();
+
+    let state = token
+        .get_account_info(&alice_meta.token_account)
+        .await
+        .unwrap();
+    assert_eq!(state.base.amount, 42);
+
+    token
+        .confidential_transfer_empty_account(
+            &alice_meta.token_account,
+            &alice,
+            &alice_meta.elgamal_keypair,
+        )
+        .await
+        .unwrap();
+}
+
+#[tokio::test]
+async fn ct_transfer() {
```

Comment:
@samkim-crypto - this is hopefully a decent template that you could start from to add tests for transfers with fee

Reply:
Great! Tests to be added soon!